src/shortestpaths/desopo-pape.jl

```diff
@@ -0,0 +1,49 @@
+"""
+    struct DEposoPapeState{T, U}
+
+An [`AbstractPathState`](@ref) designed for D`Esopo-Pape shortest-path calculations.
+"""
+struct DEsopoPapeState{T <:Real, U <: Integer} <: AbstractPathState
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+function desopo_pape_shortest_paths(g::AbstractGraph, 
+    src::Integer,
+    distmx::AbstractMatrix{T} = weights(g)) where T <: Real
+    U = eltype(g)
+    nvg = nv(g)
+    dists = fill(typemax(T), nvg)
+    parents = zeros(U, nvg)
+    state = Vector{UInt8}()
+    for i=1:nvg
+        push!(state, 2)
+    end
```

Comment:
Repeately pushing is not that performant. You could instead just write `state = fill(UInt8(2), nvg)`

Reply:
Ok!